### DISCUSSION TITLE
[Bug] log4j.yaml DefaultRolloverStrategy doesn't work

### DISCUSSION THREAD

**Discussion by wedka** [apache/pulsar#20942] [Q&A]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

Pulsar 2.10.2

### Minimal reproduce step

In the log4j.yaml，the default configuration of DefaultRolloverStrategy is:
DefaultRolloverStrategy:
          Delete:
            basePath: ${sys:pulsar.log.dir}
            maxDepth: 2
            IfFileName:
              glob: "*/${sys:pulsar.log.file}*log.gz"
            IfLastModified:
              age: 30d
However, after 30 days, the files are not deleted

### What did you expect to see?

After 30 days, the files are deleted

### What did you see instead?

After 30 days, the files are not deleted

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
@wedka would you share the files remaining in your system (with proper redact)?

`DefaultRolloverStrategy` doesn't delete logs 30 days ago, but delete logs rolling file 30 days ago. Perhaps you can use `ls -l` to show the age of all the remaining files you expect to be deleted.

**Reply by tisonkun** [reply]
This is more likely a usage issue than a bug.
